### PR TITLE
Fix several issues on Experience Index schema

### PR DESCRIPTION
**Pull request by lebaudantoine**
## Purpose

Fix several issues on Experience Index schema

## Proposal

- [x] Rename protected column name 'format'
- [x] Fix inheritance issue of table args inspired by [this workaround](https://docs.sqlalchemy.org/en/14/orm/declarative_mixins.html#combining-table-mapper-arguments-from-multiple-mixins)

Please note, that SQLModel schema might be refactored at some point when switching from SQLModel to native SQLAlchemy.

### CODE REVIEW

File: src/api/core/warren/migrations/versions/94d0308156c1_fix_xi_time_constraint.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""Fix xi time constraint
```
Comment (lebaudantoine) [inline src/api/core/warren/migrations/versions/94d0308156c1_fix_xi_time_constraint.py:1]: might be improved, to a more explicit description.

File: src/api/core/warren/migrations/versions/86ec32885672_rename_format_experience.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# revision identifiers, used by Alembic.
+revision: str = "86ec32885672"
+down_revision: Union[str, None] = "94d0308156c1"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.alter_column("experience", "format", new_column_name="technical_datatypes")
```
Comment (lebaudantoine) [inline src/api/core/warren/migrations/versions/86ec32885672_rename_format_experience.py:21]: note: will not work with mysql.

Comment (jmaupetit) [reply src/api/core/warren/migrations/versions/86ec32885672_rename_format_experience.py:21]: Why?

File: src/api/core/warren/migrations/versions/94d0308156c1_fix_xi_time_constraint.py (hunk 1 above)
Comment (jmaupetit) [reply src/api/core/warren/migrations/versions/94d0308156c1_fix_xi_time_constraint.py:1]: Should be fine. As long as the commit message is explicit, we are fine.

**Review by jmaupetit — APPROVED** [review]

File: src/api/core/warren/migrations/versions/86ec32885672_rename_format_experience.py (hunk 2 above)
Comment (lebaudantoine) [reply src/api/core/warren/migrations/versions/86ec32885672_rename_format_experience.py:21]: [based on this discussion](https://stackoverflow.com/questions/22599391/how-to-use-alter-column-in-alembic#comment96733376_47184837)

**Pull request merged**